spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -387,6 +387,19 @@ will run in the context of an LRA and controls whether or not:
 - to throw an exception if the method returns particular HTTP status codes.
 - to enlist the class as a participant of the LRA.
 
+Some of the above defined LRA operations are performed before the execution proceeds
+to the invoked `@LRA` annotated business method. If any of these LRA operations
+prohibits the execution of the business method, the LRA implemenation MUST return
+one of the following status codes to user invoking the LRA annotated method. This
+means that in these conditions the business method _is not_ invoked:
```

Comment:
What added value does this paragraph provide? The two bullets below say everything that needs to be said. I had to read this paragraph a couple of times just to make sure I was not missing any subtle points (so adds complexity).

Reply:
I believe that we need to introduce what operations we are talking about and to put the emphasis on the method not being executed. Without this paragraph maybe the user will understand that his business method needs to return these HTTP status codes? I don't think it's adding complexity, rather it's adding context.

Reply:
I think it's good to define that these error codes relates to case when the business method has not been executed yet.
From that perspective it's valuable to introduce those two points with some text.
I agree in way that the text is a bit hard to read. Maybe just one sentence which emphasizes the key point would be better?

Reply:
I can remove the last sentence but that the emphasis that I wanted to put there for users. Other text is important.

Reply:
I'm not able to create some nicer rephrasing of the paragraph so that way it would be good from my perspective.

Reply:
@ochaloup The two bullets below (turned into two paragraphs) provides all of the information that the user needs about what error codes are returned without the method having to execute. The above paragraph adds nothing to his understanding, on the contrary it can give the reader the impression that he has missed some nuance and therefore obfuscates the meaning of the two bullets.

Reply:
@mmusgrov I see that this argument is a matter of taste but there are two opinions of non-native speakers which think that this paragraph (excluding last sentence) is adding value to the reader. 

I don't see how this reads better:

```
The annotation MUST be applied to JAX-RS annotated methods, classes, superclasses or interface methods otherwise
it has no effect. It determines whether or not the JAX-RS method
will run in the context of an LRA and controls whether or not:

- any incoming context should be suspended and if so if a new one should be started.
- to start a new LRA.
- to end any LRA context when the method ends.
- to throw an exception if there should be an LRA present on method entry.
- to throw an exception if the method returns particular HTTP status codes.
- to enlist the class as a participant of the LRA.

If the incoming context has already ended or does not exist the method will 
immediately return 410 Gone HTTP status code.

Similarly, if any of the preconditions listed in the Type element
(see <<source-LRA,LRA.Type>>) of the LRA annotation are not met then the method
will immediately return 412 Precondition Failed HTTP status code.
```

You yourself argued that I didn't define what operations I was describing. I can see where I would find different meanings if I didn't know what the paragraphs talk about.

Reply:
@mmusgrov @xstefank For me it seemed that it's good to highlight that the checks are executed before the business method and they cause the method won't bet invoked. It's my preference but I do not have a strong argument if it's necessary.

Reply:
@xstefank the text you just quoted looks fine (in fact it is what I have been suggesting). What are you saying here?

@ochaloup The text I suggested does say that the checks are executed before the business method runs. To quote: "... the method will immediately return ...". Are you interpreting those 5 words to mean something other than to not execute any business logic? If you are then would "... the method will not execute and a 410 Gone HTTP status code will be returned back to the caller" be preferable?

Reply:
I am not labouring this point to be obstructive. I have said on a number of occasions that adding too much text or saying a similar thing in multiple places or adding superfluous explanations is making our spec quite difficult to parse.

Reply:
ok, I see that this is going nowhere and I really don't have any more time or arguments on this.

Reply:
You still have the text that I said was duplicating the bullets in the spec, except now its in the javadoc. But this is fine since the javadoc is allowed to repeat what is said in the main document and it is useful that the duplication is worded differently.